### PR TITLE
Release v1.2.2 [full merge, no squash]

### PR DESCRIPTION
**Pull request by kaushikb11**
## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,48 @@ All notable changes to this project will be documented in this file.
...
+- Fixed incorrect usage of `detach()`, `cpu()`, `to()` ([#6216](https://github.com/PyTorchLightning/pytorch-lightning/pull/6216))
+
+
+- Fixed LBFGS optimizer support which didn't converge in automatic optimization ([#6147](https://github.com/PyTorchLightning/pytorch-lightning/pull/6147))
+
+
+- Prevent `WandbLogger` from dropping values ([#5931](https://github.com/PyTorchLightning/pytorch-lightning/pull/5931))
+
+
+- Fixed error thrown when using valid distributed mode in multi node ([#6297](https://github.com/PyTorchLightning/pytorch-lightning/pull/6297)
```
Comment (carmocca) [inline CHANGELOG.md:47]: Remove the spaces between the lines

Also the empty sections

**Review by carmocca — COMMENTED** [review]
1b498d1 (#6075)

is missing

**Comment by kaushikb11** [comment]
@carmocca #6075 is not missing. Check the first update in Changelog.

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: tests/overrides/test_data_parallel.py [hunk 2]
```diff
@@ -147,6 +147,7 @@ def training_step(self, batch, batch_idx):
     model = TestModel().to(device)
     model.trainer = MagicMock()
     model.trainer._running_stage = RunningStage.TRAINING
+    model.running_stage = RunningStage.TRAINING
```
Comment (awaelchli) [inline tests/overrides/test_data_parallel.py:150]: why was this added?
the LightningModule should have a property called `running_stage`
why is it not there?

**Review by awaelchli — CHANGES_REQUESTED** [review]

**Review by awaelchli — APPROVED** [review]

File: tests/overrides/test_data_parallel.py (hunk 2 above)
Comment (carmocca) [reply tests/overrides/test_data_parallel.py:150]: Maybe the latest changes should be undone and we should cherry-pick https://github.com/PyTorchLightning/pytorch-lightning/commit/0456b4598f5f7eaebf626bca45d563562a15887b instead

Comment (awaelchli) [reply tests/overrides/test_data_parallel.py:150]: I had a brief confusion, but then remembered that this only contains bugfix commits.
Will 1.2.3 be based on master again or continue off of 1.2.2?

Comment (carmocca) [reply tests/overrides/test_data_parallel.py:150]: Continue off 1.2.2

**Review by Borda — CHANGES_REQUESTED** [review]
can you check what is the last commit, seems like merging master...
https://github.com/kaushikb11/pytorch-lightning

**Review by Borda — APPROVED** [review]
ok, seems fine after clean

**Comment by Borda** [comment]
@lexierule mind do merge without squash as we did last time?
in setting allow "rebase merge" - come here, with admin rebase merge - in setting disable "rebase merge"
